framework/tests/h1h2checker/Issue282.java

```diff
@@ -0,0 +1,15 @@
+import org.checkerframework.framework.testchecker.h1h2checker.quals.*;
+
+public class Issue282 {
+    // :: error: (super.invocation.invalid) :: warning: (inconsistent.constructor.type)
+    @H1S1 Issue282() {}
+
+    public class Inner {
+        Inner(@H2S2 Issue282 Issue282.this) {}
+        ;
+    }
+
+    public void test1() {
+        Inner inner = new Issue282().new Inner();
```

Comment:
Shouldn't there be an expected error here? The `new Issue282()` has type `@H1S1 Issue282`, which is incompatible with the receiver of the `Inner` constructor, which is `@H2S2 Issue282`.

Add an additional constructor to `Inner` where the test passes.

Reply:
It should be, however this commit was kind of a draft version since I did not finish the type checking part.